index.js

```diff
@@ -156,15 +156,22 @@ RedirectableRequest.prototype.setTimeout = function (msecs, callback) {
     this.on("timeout", callback);
   }
 
+  function setSocketTimeout(socket) {
+    socket.removeListener("timeout", socket.destroy);
```

Comment:
This `socket.removeListener("timeout", socket.destroy);` is here because this method gets called twice for the first request. This way we avoid adding redundant `timeout` listeners to the first request.